documentation/src/main/asciidoc/topics/marshalling.adoc

```diff
@@ -1,82 +1,55 @@
+for user types if required.
 
-* cache is configured to be a local cache _and..._
-* cache is not configured with lazy serialization _and..._
-* cache is not configured with any write-behind cache store
+== Marshaller Implementations
+[[protostream_cm_config]]
+=== ProtoStream (Default)
+The default marshaller for {brandname} is <<protostream,ProtoStream>>, which marshalls data as link:https://developers.google.com/protocol-buffers[Protocol Buffers]. This is a platform independent format
+that utilises schemas to provide a structured representation of your Java objects in a manner that can be easily evolved
+over time whilst maintaining backwards compatibility.
 
-If either of these options is true, key/value pairs in the cache will need to be marshalled and currently they require to either to extend `java.io.Serializable` or `java.io.Externalizable`.
+{brandname} directly integrates with the ProtoStream library by allowing users to configure implementations of the ProtoStream
+SerializationContextInitializer interface. These implementations are then used to initialise the various SerializationContext's used by
+{brandname} for marshalling, allowing custom user objects to be marshalled for storage and cluster communication.
+More details on how to generate/implement your own SerializationContextInitailizers can be found <<protostream_usage,here>>.
 
-TIP: Since {brandname} 5.0, marshalling non-Serializable key/value objects are supported as long as users can to provide meaningful Externalizer implementations for these non-Seralizable objects.
+SerializationContextInitailizers can be configured as follows:
 
-If you're unable to retrofit Serializable or Externalizable into the classes whose instances are stored in {brandname}, you could alternatively use something like link:http://x-stream.github.io/[XStream] to convert your Non-Serializable objects into a String that can be stored into {brandname}. The one caveat about using XStream is that it slows down the process of storing key/value objects due to the XML transformation that it needs to do.
-
-[[store_as_binary]]
-=== Store As Binary
-Store as binary enables data to be stored in its serialized form. This can be useful to achieve lazy deserialization, which is the mechanism by which {brandname} by which serialization and deserialization of objects is deferred till the point in time in which they are used and needed. This typically means that any deserialization happens using the thread context class loader of the invocation that requires deserialization, and is an effective mechanism to provide classloader isolation. By default lazy deserialization is disabled but if you want to enable it, you can do it like this:
-
-* Via XML at the Cache level, either under `<*-cache />` element:
+*  Via XML at the CacheManager level, under `<cache-manager />` element:
```

Comment:
Instead of bullets, use the heading markup like this:

".Declarative procedure "
".Programmatic procedure"

for example, https://raw.githubusercontent.com/infinispan/infinispan/master/documentation/src/main/asciidoc/topics/proc_using_jgroups_external.adoc

Reply:
That definitely looks better. I've updated the other sections in `marshalling.adoc` that I didn't create as well.